rpc/Base.ts

```diff
@@ -8,41 +9,84 @@ export type ListenerCb<IngressMessage_ extends M.IngressMessage = M.IngressMessa
 
 export type StopListening = () => void;
 
-export class RpcError extends Error {
-  constructor(readonly inner: M.ErrMessage) {
-    super();
-  }
-}
+export abstract class RpcClient<RpcError extends E.RpcError> {
```

Comment:
Why is this generic on the error?

Reply:
My thinking is that we'll want beacon-specific error types (web sockets can error out differently from Smoldot client instances).